### PR TITLE
fix: array_resize null fix

### LINKED ISSUE #13200: null value causes panic in array_resize
### Describe the bug

Panic in function array_resize when being passed a null value.

### To Reproduce

```
DataFusion CLI v42.1.0
> select array_resize(arrow_cast(NULL, 'List(Int8)'), 1);
thread 'main' panicked at /home/eejbyfeldt/.cargo/registry/src/index.crates.io-6f17d22bba15001f/arrow-data-53.2.0/src/transform/primitive.rs:31:43:
range end index 1 out of range for slice of length 0
```

### Expected behavior

It should return a null value instead.

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by jonathanc-n**
## Which issue does this PR close?

Closes #13200 .

## What changes are included in this PR?
Added null check for array_resize

### CODE REVIEW

File: datafusion/sqllogictest/test_files/array.slt [hunk 1]
```diff
@@ -7013,6 +7013,12 @@ select array_resize(arrow_cast([[1], [2], [3]], 'LargeList(List(Int64))'), 10, [
 ----
 [[1], [2], [3], [5], [5], [5], [5], [5], [5], [5]]
 
+# array_resize null value
+query ?
+select array_resize(arrow_cast(NULL, 'List(Int8)'), 1);
+----
+NULL
```
Comment (jatin510) [inline datafusion/sqllogictest/test_files/array.slt:7020]: shouldn't it be 
[NULL] 

@jonathanc-n

Comment (findepi) [reply datafusion/sqllogictest/test_files/array.slt:7020]: i think it's fine. the query creates no array (besides null), so i wouldn't expect 1-element array in the output.

File: datafusion/functions-nested/src/resize.rs [hunk 2]
```diff
@@ -134,6 +136,23 @@ pub(crate) fn array_resize_inner(arg: &[ArrayRef]) -> Result<ArrayRef> {
         return exec_err!("array_resize needs two or three arguments");
     }
 
+    let array = &arg[0];
+
+    // Checks if entire array is null
+    if array.null_count() == array.len() {
```
Comment (findepi) [inline datafusion/functions-nested/src/resize.rs:142]: `array` is an array of arrays, right?
what happens if some values are null and some other are not?

For example a query processes some null and non-null arrays
```
> SELECT a FROM (VALUES (make_array(1)), (arrow_cast(NULL, 'List(Int32)')), (make_array(3))) t(a);
+-----+
| a   |
+-----+
| [1] |
|     |
| [3] |
+-----+
```

currently, array_resize converts second array (being null) into an empty array.

```
> SELECT array_resize(a, 1) FROM (VALUES (make_array(1)), (arrow_cast(NULL, 'List(Int32)')), (make_array(3))) t(a);
+----------------------------+
| array_resize(t.a,Int64(1)) |
+----------------------------+
| [1]                        |
| []                         |
| [3]                        |
+----------------------------+
```

File: datafusion/sqllogictest/test_files/array.slt (hunk 1 above)
Comment (jatin510) [reply datafusion/sqllogictest/test_files/array.slt:7020]: ` Resizes the list to contain size elements`
gotcha !

Thanks @findepi

File: datafusion/sqllogictest/test_files/array.slt [hunk 3]
```diff
@@ -7013,6 +7013,12 @@ select array_resize(arrow_cast([[1], [2], [3]], 'LargeList(List(Int64))'), 10, [
 ----
 [[1], [2], [3], [5], [5], [5], [5], [5], [5], [5]]
 
+# array_resize null value
+query ?
+select array_resize(arrow_cast(NULL, 'List(Int8)'), 1);
```
Comment (jayzhan211) [inline datafusion/sqllogictest/test_files/array.slt:7018]: Can we also add columnar test?

File: datafusion/functions-nested/src/resize.rs (hunk 2 above)
Comment (jonathanc-n) [reply datafusion/functions-nested/src/resize.rs:142]: Thanks for pointing that out, I overlooked that.

**Review by jayzhan211 — APPROVED** [review]
👍

**Comment by jonathanc-n** [comment]
I kept two null checks in the array_resize function: one in the array_resize_inner function and one in the general_list_resize function. The first null check is needed for the original problem of handling cases like SELECT array_resize(arrow_cast(NULL, 'List(Int8)'), 1);. This scenario can't be checked by the second null check because the array has a length of zero. With a length of zero, it conflicts with the null builder, as the null counts as a length of one, which isn't the expected value when constructing the result. This is due to the null array and the data array needing to have matching lengths.

**Comment by jayzhan211** [comment]
```rust
error[E0599]: no method named `offsets` found for reference `&GenericByteViewArray<StringViewType>` in the current scope
   --> /Users/bytedance/.cargo/registry/src/index.crates.io-6f17d22bba15001f/arrow-string-53.2.0/src/length.rs:142:50
    |
142 |             Ok(bit_length_impl::<Int64Type>(list.offsets(), list.nulls()))
    |                                                  ^^^^^^^
    |
help: there is a method `offset` with a similar name
    |
142 |             Ok(bit_length_impl::<Int64Type>(list.offset(), list.nulls()))
    |                                                  ~~~~~~

For more information about this error, try `rustc --explain E0599`.
error: could not compile `arrow-string` (lib) due to 1 previous error
warning: build failed, waiting for other jobs to finish...
```

I got this error when running query in this branch 🤔

**Comment by jatin510** [comment]
lgtm  @alamb

**Comment by jayzhan211** [comment]
Thanks @jonathanc-n @findepi @jatin510

**Pull request merged**